sct-commons/src/test/java/org/lfenergy/compas/sct/commons/testhelpers/ControlBlockNetworkingCsvHelper.java

```diff
@@ -0,0 +1,202 @@
+// SPDX-FileCopyrightText: 2022 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons.testhelpers;
+
+import com.opencsv.bean.ColumnPositionMappingStrategy;
+import com.opencsv.bean.CsvBindByPosition;
+import com.opencsv.bean.CsvToBeanBuilder;
+import com.opencsv.enums.CSVReaderNullFieldIndicator;
+import lombok.*;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.lfenergy.compas.scl2007b4.model.TCompasICDHeader;
+import org.lfenergy.compas.scl2007b4.model.TCompasIEDRedundancy;
+import org.lfenergy.compas.scl2007b4.model.TCompasIEDType;
+import org.lfenergy.compas.scl2007b4.model.TDurationInMilliSec;
+import org.lfenergy.compas.sct.commons.dto.ControlBlockNetworking;
+import org.lfenergy.compas.sct.commons.scl.ied.ControlBlockAdapter;
+import org.lfenergy.compas.sct.commons.scl.ied.IEDAdapter;
+import org.lfenergy.compas.sct.commons.util.ControlBlockEnum;
+
+import java.io.Reader;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * This class is an implementation example for interface ControlBlockNetworking.
+ * It relies on a CSV file.
+ * First columns of the CSV file are the criteria to match the ControlBlock,
+ * last columns are the return values for the template for the matched ControlBlock.
+ */
+public class ControlBlockNetworkingCsvHelper implements ControlBlockNetworking {
+
+    private static final char SEPARATOR = ';';
+    private static final int HEADER_LINES = 5;
+    private static final int MAX_VLAN_ID = 0x0FFF;
+    private static final int MAX_VLAN_PRIORITY = 7;
+
+    private final Map<Criteria, Template> templates;
+
+
+    public ControlBlockNetworkingCsvHelper(Reader csvSource) {
+        templates = readCsvFile(csvSource);
+    }
+
+    @Override
+    public Template getNetworkTemplate(ControlBlockAdapter controlBlockAdapter) {
```

Comment:
Why delegate this verification to user? Wile it is a compas specification

Reply:
It is not a verification. The user has to provide data for each ControlBlock : vlanId, vlanPriority, minTime, maxTime.
COMPAS cannot guess these data.
If we use an interface like this, each user can decide it's own algorithm to provide vlanId, vlanPriority, minTime, maxTime for each ControlBlock.
RTE has an algorithm based on an Excel file, but this method is specific to RTE and will be implemented in RCONF (In fact ControlBlockNetworkingCsvHelper.java is an implementation example that will go in RCONF when RCONF will want to use this service).

We should not put RTE Excel files in COMPAS (I dit it for an other issue but it was a mistake and I created a technical issue to refactor it : RSR-685)

Reply:
Ok for Excel files not be put in compas. But in the issue there are specific treatment applied to entries (comming from excel for RTE but for other should just give the information as asked input of method) so i think that unless implementation of interface you should include those check into compas.

Reply:
OK, so I moved `ControlBlockNetworkingCsvHelper` to the production classes, in `Utils` package, and I added a new signature for method `ExtRefService#configureNetworkForAllControlBlocks` that take the CSV file as parameter (`csvSource`) like this : 

```java
    public static SclReport configureNetworkForAllControlBlocks(SCL scd, Reader csvSource,
                                                                long appIdMin, long appIdMax, String macAddressMin, String macAddressMax, ControlBlockEnum controlBlockEnum) {
        ControlBlockNetworkSettings controlBlockNetworkSettings = new ControlBlockNetworkSettingsCsvHelper(csvSource);
        return configureNetworkForAllControlBlocks(scd, controlBlockNetworkSettings, appIdMin, appIdMax, macAddressMin, macAddressMax, controlBlockEnum);
    }
```